src/Strategy/Node/YarnV1/YarnLock.hs

```diff
@@ -173,8 +175,8 @@ missingResolvedVersionErrorMsg key =
 pairToPackage :: YL.PackageKey -> YL.Package -> YarnV1Package
 pairToPackage key pkg = YarnV1Package (extractFullName key) (YL.version pkg)
 
-firstKey :: (NE.NonEmpty a, b) -> (a, b)
-firstKey (neList, pkg) = (NE.head neList, pkg)
+allkeys :: (NE.NonEmpty a, b) -> (a, [a], b)
+allkeys (neList, pkg) = (NE.head neList, NE.toList neList, pkg)
```

Comment:
[suggest]: I think naming and type might need to change here, in it's current state we're kind of just putting a band-aid on it.  A complete fix (i.e.: doing it right the first time) would probably look more like this:
```suggestion
extractPkgTriple :: (NE.NonEmpty YL.PackageKey, YL.Package) -> (Text, [YL.PackageKey], YL.Package)
extractPkgTriple (neList, pkg) = (extractFullName $ NE.head neList, NE.toList neList, pkg)
```

Reply:
sigh - I def needed a coffee that day. Applied all suggestions!